main/webapp/modules/core/scripts/project/history-panel.js

```diff
@@ -288,9 +288,34 @@ HistoryPanel.prototype._showApplyOperationsDialog = function() {
   
   elmts.dialogHeader.html($.i18n('core-project/apply-operation'));
   elmts.or_proj_pasteJson.html($.i18n('core-project/paste-json'));
+
+  elmts.operationJsonButton.on('click', async function() {
+    const input = document.createElement('input');
+    input.type = 'file';
+    input.accept = '.json'; // optional file types
+    input.onchange = async function() {
+      const file = input.files[0];
+      const reader = new FileReader();
+      reader.onload = function(e) {
+        const fileContent = JSON.parse(e.target.result);
+        const textAreaElement = document.querySelector('#textareaId')
```

Comment:
```suggestion
        const textAreaElement = elmts.textarea;
```
This would avoid looking up the textarea via a global identifier (which could easily conflict with another element, since the id is pretty generic) and instead select the one in the current dialog. Please check if it works though!

Reply:
Unsuccessful, brings about error so I will maintain how it was initially.

Reply:
Have you tried with `elmts.textarea[0]` instead? Looking at the rest of the code, it seems that it could be better like that.